src/evaluator.ts

```diff
@@ -76,9 +80,19 @@ export class Evaluator {
     bucketBy: string,
     percentage: number,
   ): boolean {
-    const property = this.getAttrValue(target, bucketBy);
+    let property = this.getAttrValue(target, bucketBy);
     if (!property) {
-      return false;
+      const old_bucketBy = bucketBy;
+      bucketBy = 'identifier';
+      property = this.getAttrValue(target, bucketBy);
+      if (!property) {
+        return false;
+      }
+      warnBucketByAttributeNotFound(
```

Comment:
This is becoming a bit of a wall of code. A few line breaks around the `if` will make it a lot easier to visually scan.
```suggestion
      property = this.getAttrValue(target, bucketBy);
      
      if (!property) {
        return false;
      }
      
      warnBucketByAttributeNotFound(
```